Dockerfile

```
@@ -0,0 +1,21 @@
+FROM python:3.11
```

I suggest using a `slim` base image, it has the same glibc-based environment, but includes less irrelevant tools out-of-the-box (~160mb vs 1gb).

```suggestion
FROM python:3.11-slim
```

Also, is there a reason why we don't use Python v3.12, given that the CI runs tests for it?